### PR TITLE
Faster reset for 'nn' modules.

### PR DESCRIPTION
**Pull request by clementfarabet**
Gents,

I modified all the reset() methods in 'nn' to use our new vectorial uniform/normal methods. These work on both CPU and GPU, and are much faster than the apply we currently have.

I didn't push it to the master right away because it probably produces different initializations for a a same seed. Is that OK for everyone?

### CODE REVIEW

**Comment by clementfarabet** [comment]
Ping: anyone interested in this?

When initializing a fat nn.Linear() layer, say nn.Linear(100000, 1000), it takes a minute to reset the weights currently. With this patch, it's a second.

**Comment by andresy** [comment]
I am scared to have different seeds than before :)
Ach. With a big warning "WE CHANGED THE SEED" in the commit log?

**Comment by clementfarabet** [comment]
Right... I had forgotten about this problem. Good point. Is anybody else scared by this? Maybe we should wait then.

**Comment by jucor** [comment]
The big warning and a reference in the next Changelog to the commit right before the change sounds good for me.
Maybe even a git tag "OriginalSeed" ? Reproduction of previous results can then be done by going back to tagged commit.

**Comment by clementfarabet** [comment]
Sounds good to me!  

Since we have rocks now, any Torch version that's <= 7.0.3 will have the old seed, any version that's >= 7.0.4 will us the new seed.

By the way, we should have a little log page that summarizes what's new in each rock version (I'm creating one every week or every 2 weeks, depending on the amount of new stuff in the master branch).  

Clément

**Comment by koraykv** [comment]
How about if we seek for a variable named nn.oldSeed being defined and if found, use the old way of initialising. In the reset function, we use the new way, but if it is defined, run the old code.

**Comment by jucor** [comment]
Best of both worlds, Koray :-)

**Comment by andresy** [comment]
could you undo this merge ASAP? it modifies a lot of things (including messing up the torch paths), which have nothing to do with nn reset

**Comment by clementfarabet** [comment]
Done. Koray I followed your advice, and kept all the legacy reset code, conditioned on nn.oldSeed. If that variable is defined, everything goes back to the old reset style.

**Comment by clementfarabet** [comment]
Ah crap, it's not this merge, it's the luarocks one, sorry about that. Will undo it.

**Comment by clementfarabet** [comment]
OK I've reverted the paths, and re-created the luarocks branch.
